### PR TITLE
Feat/stagingBuildTimes

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem

BE for staging build status. Corresponding [FE pr](https://github.com/isomerpages/isomercms-frontend/pull/1635) 

Closes [insert issue #]

## Solution

Get the latest commit as SOT. 

## Tests
done in fe

### CODE REVIEW

File: src/types/stagingBuildStatus.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const statusStates = {
```
Comment (dcshzj) [inline src/types/stagingBuildStatus.ts:1]: nit: this should be fully capitalised with snake case

File: src/services/identity/SitesService.ts [hunk 2]
```diff
@@ -536,6 +541,26 @@ class SitesService {
...
+      return errAsync(new NotFoundError())
+    }
+    return this.getBySiteName(siteName)
+      .andThen((site) =>
+        this.deploymentsService.getStagingSiteBuildStatus(
+          site.id.toString(),
+          isReduceBuildTimesWhitelistedRepo(growthbook)
+        )
+      )
+      .andThen((status) => okAsync({ status }))
```
Comment (dcshzj) [inline src/services/identity/SitesService.ts:562]: nit: should use `.map` instead

File: src/services/infra/InfraService.ts [hunk 3]
```diff
@@ -37,6 +37,7 @@ import {
   SiteLaunchStatusObject,
 } from "@root/types/siteInfo"
 import { SiteLaunchMessage } from "@root/types/siteLaunch"
+import { isReduceBuildTimesWhitelistedRepo } from "@root/utils/growthbook-utils"
```
Comment (dcshzj) [inline src/services/infra/InfraService.ts:40]: nit: unused?

**Review by dcshzj — APPROVED** [review]
Non-blocking comments

File: src/services/identity/DeploymentsService.ts [hunk 4]
```diff
@@ -259,6 +262,63 @@ class DeploymentsService {
...
+    siteId: string,
+    isRepoWhiteListedForBuildRed: boolean
+  ): ResultAsync<BuildStatus, NotFoundError | AmplifyError> =>
+    fromPromise(
+      this.deploymentsRepository.findOne({
+        where: {
+          siteId,
+        },
+      }),
+      () => new NotFoundError("Site has not been deployed!")
```
[the remark this answers is not in the document]
Comment (kishore03109) [reply src/services/identity/DeploymentsService.ts:276]: wait no in this case the deployments row doesnt exist, so it is the case that the site has not been deployed!

**Pull request merged**